File: impares.m
pkg load database;

clc; 
clear all; 
close all;

retry= true;

while retry
  try
    fprintf("Elija una opcion: 1. Mostrar los numeros impares,   2. Historial: ");
    y=(input ("Opcion a elegir: "));
    if y>2||y==0||y<0
      fprintf("Numero erroneo, vuelva a ingresar: \n");
    endif
    
    if y==1
      imp=1;
      for x=1:2:100
        V(imp)=x;
        imp=imp+1;
      endfor
      disp(V)
     %Conectar con PGADMIN
      res=imp-1;
      linearecta=mat2str(V);
      params=cell(1,2);
      params{1,2}=res;
      params{1,1}=linearecta;


      conn = pq_connect(setdbopts('dbname','tarea1','host','localhost',
      'port','5432','user','postgres','password','jblue5queso'));
      N=pq_exec_params(conn, "insert into impares(res, cantidad) values($1,$2);",params); %insertar datos en la tabla

      %Texto
      save('impares.txt','V','-ascii');
      retry=false;
    endif
     if y==2
       conn = pq_connect(setdbopts('dbname','tarea1','host','localhost',
       'port','5432','user','postgres','password','jblue5queso'));
       N=pq_exec_params(conn, "Select*from impares;"); %historial de la tabla
       disp(N)
       retry=false;
     endif
      catch
      fprintf("Solo se permiten valores numericos, vuelava a selecionar %d \n");
      msg = lasterror.message;
      fprintf(msg);
    end_try_catch
  endwhile

  
  

File: Impares.py
import psycopg2
import math
from tabulate import tabulate
def  post(x,y):
    try:
        conexion = psycopg2.connect(
            host = "localhost",
            port = "5432",
            user = "postgres",
            password = "jblue5queso",
            dbname = "tarea1"
            )
    except:
        print("Sin Conexion Exitosa\n")
            
    cursor = conexion.cursor()
    cursor.execute("INSERT INTO impares(res, cantidad) VALUES (%s, %s);", (x, y))
    conexion.commit()
    cursor.close()
    conexion.close()

def ret():
    try:
        conexion = psycopg2.connect(
            host = "localhost",
            port = "5432",
            user = "postgres",
            password = "jblue5queso",
            dbname = "tarea1"
            )
    except:
        print("Sin Conexion Exitosa\n")
    
    cursor = conexion.cursor()
    cursor.execute("SELECT * from impares;")
    print(tabulate(cursor, headers=["ide", "res", "cantidad"], tablefmt="orgtbl", numalign ="center"))

print("Los numeros del 1 al 100 que son impares: \n")
mat=[]
while True:
    try:
        print("Introduce el numero de la operacion a realizar (1.Numeros impares (1-100), 2.Historial, 3.Salir): ")
        x=int(input("Elija:\n"))
        if x>3 or x<=0:
            print("Valor ingresado fuera del rango del menu.\n")
        elif x==1:
            for y in range(1,100,2):
                mat.append(y)
            print("Los numero impares del 1 al 100 son: \n ",mat)
            post(mat,len(mat))
        elif x==2:
            ret()
        elif x==3:
            break

    except Exception as es:
        print("\n Ingreso un caracter y no un numero, vuelva a intentar:\n")
        print(repr(es))

exit(0)